api_app/analyzers_manager/observable_analyzers/haveibeenpwned.py

```diff
@@ -0,0 +1,48 @@
+import requests
+
+from api_app.analyzers_manager import classes
+from api_app.exceptions import AnalyzerRunException
+from tests.mock_utils import MockResponse, if_mock_connections, patch
+
+
+class HaveIBeenPwned(classes.ObservableAnalyzer):
+    base_url: str = "https://haveibeenpwned.com/api/v3/breachedaccount/"
+
+    def set_params(self, params):
+        self.truncate_response = params.get("truncate_response", True)
+        self.include_unverified = params.get("include_unverified", True)
+        self.domain = params.get("domain", None)
+        self.__api_key = self._secrets["api_key_name"]
+
+    def run(self):
+        params = {
+            "truncateResponse": self.truncate_response,
+            "includeUnverified": self.include_unverified,
+        }
+        if self.domain:
+            params["domain"] = self.domain
```

Comment:
if you check the Coverage results, your PR reduces the overall coverage. We need that the test for this analyzer executes this line too. To do that, you need to add this optional parameter here: https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/__init__.py#L89